docs/usage/search.md

```diff
@@ -1,3 +1,187 @@
+
+    # Include only .json and .txt files
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=your_query&include=*.json&include=*.txt"
+
+    # Exclude 'node_modules' in the root directory
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=your_query&exclude=/node_modules"
+
+    # Exclude 'node_modules' on all levels
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=your_query&exclude=**/node_modules"
+
+    # Combine multiple parameters
+    curl -X GET "http://localhost:8080/projects/{projectId}/search?type=content&query=your_query&include=*.json&include=*.txt&exclude=**/node_modules&showHidden"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+These additional parameters allow you to fine-tune your search to specific file types or directories, making it easier to find exactly what you're looking for in your project.
+
+## File Search
+
+File search helps you find files within your project based on their names or paths. This is particularly useful when you're looking for specific files or want to filter files based on certain patterns.
+
+### Basic Usage
+
+To search for files:
+
+=== "curl"
+
+    ```bash
+    curl -X GET "http://localhost:8080/projects/{projectId}/files"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+### Filtering Files
+
+You can use `include` and `exclude` parameters to filter the search results:
+
+=== "curl"
+
+    ```bash
+    # Include only Python files
+    curl -X GET "http://localhost:8080/projects/{projectId}/files?include=*.py"
+
+    # Exclude test files
+    curl -X GET "http://localhost:8080/projects/{projectId}/files?exclude=test_*.py"
+
+    # Combine include and exclude
+    curl -X GET "http://localhost:8080/projects/{projectId}/files?include=*.py&exclude=test_*.py"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+## Symbol Search
+
+Symbol search allows you to find specific symbols (like functions, classes, or variables) within your project. This is extremely helpful when you're trying to locate specific code elements without knowing their exact file location.
+
+### Basic Usage
+
+To search for symbols:
+
+=== "curl"
+
+    ```bash
+    curl -X GET "http://localhost:8080/projects/{projectId}/symbols?type=symbol&query=your_symbol_name"
+    ```
+
+=== "python"
+
+    ```python
+    # Coming soon
+    ```
+
+### Advanced Options
+
+You can customize the symbol search with additional parameters:
+
+- `limit`: Specify the maximum number of results (default is 10, max is 100)
+- Include or exclude specific symbol types. By default, the search will exclude fields and variables.
+
+=== "curl"
+
+    ```bash
+    # Limit results to 20
+    curl -X GET "http://localhost:8080/projects/{projectId}/symbols?type=symbol&query=your_symbol_name&limit=20"
```

Comment:
A bit late but do we really need `type=symbol` in `.../{projectId}/symbol` path.